### PR TITLE
feat: add release workflows

### PR DESCRIPTION
**Pull request by akarui-hikarii**

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -32,17 +35,42 @@ jobs:
...
+        run: |
+          git config user.name github-actions
+          git config user.email github-actions@github.com
 
       - name: Release
         env:
           CARGO_REGISTRY_TOKEN: ${{ secrets.CARGO_REGISTRY_TOKEN }}
         run: cargo smart-release -ue
+
+      # Leaving bevy comments here for us to discuss. Do we want to start tagging our pre-releases with `dev`?
```
Comment (akarui-hikarii) [inline .github/workflows/release.yml:48]: @gmpinder what do you think about using dev tag and then when we release, remove the dev tag

Comment (gmpinder) [reply .github/workflows/release.yml:48]: I'd be down to do that

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -32,17 +35,42 @@ jobs:
...
-          tool: cargo-smart-release
+      - name: Git setup
+        run: |
+          git config user.name github-actions
+          git config user.email github-actions@github.com
 
       - name: Release
         env:
           CARGO_REGISTRY_TOKEN: ${{ secrets.CARGO_REGISTRY_TOKEN }}
         run: cargo smart-release -ue
```
Comment (gmpinder) [inline .github/workflows/release.yml:46]: You'll want to delete this

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -18,10 +19,12 @@ jobs:
...
           fetch-depth: 0
-      - name: Git setup
-        run: |
-          git config user.name github-actions
-          git config user.email github-actions@github.com
+
+      - uses: chainguard-dev/actions/setup-gitsign@main
+
+      - name: Install cargo-release
+        run: cargo install cargo-release
```
Comment (gmpinder) [inline .github/workflows/release.yml:26]: You might want to move this after the cache step

File: .github/workflows/post-release.yml [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+            jq --raw-output '.packages | .[] | select(.name == "blue-build").version')
+
+          # Sanity check: current version should be 0.X.Y
+          if ! grep -q '^0\.[0-9]\+\.[0-9]\+$' <<< "${current_version}"; then
+            echo "Invalid version (not in 0.X.Y format): ${current_version}"
+            exit 1
+          fi
+
+          minor_version=$(sed 's/^0\.\([0-9]\+\).*/\1/' <<< "${current_version}")
+          next_version=0.$((minor_version + 1)).0-dev
```
Comment (gmpinder) [inline .github/workflows/post-release.yml:38]: Just a note, we'll need to update this later to account for the major version too as this seems to only be concerned about versions `0.x.x`

**Comment by gmpinder** [comment]
If cargo release doesn't handle publishing to crates.io, we'll need to update the tag workflow to do that or make it part of the post-release process

**Comment by akarui-hikarii** [comment]
Cargo release should push to xrates. _should_

File: .github/workflows/post-release.yml (hunk 4 above)
Comment (akarui-hikarii) [reply .github/workflows/post-release.yml:38]: noted. i think for bevy they havent release a 1.0.0 version yet so

File: .github/workflows/release.yml [hunk 5]
```diff
@@ -32,17 +32,40 @@ jobs:
             ~/.cargo/registry/cache/
           key: ${{ runner.os }}-cargo-build-stable-${{ hashFiles('**/Cargo.toml') }}
 
-      - uses: chainguard-dev/actions/setup-gitsign@main
+      - name: Install cargo-release
+        run: cargo install cargo-release
 
-      # Pulling rust toolchain & install linux dependencies
-      - uses: dtolnay/rust-toolchain@stable
```
Comment (gmpinder) [inline .github/workflows/release.yml:38]: I think we might need this?

Comment (akarui-hikarii) [reply .github/workflows/release.yml:38]: i dont know. bevy doesnt have it so im curious

**Review by gmpinder — APPROVED** [review]

**Pull request merged**